### ISSUE TITLE
Support generation of path array

### ISSUE THREAD

**Issue by syxov** [dsherret/ts-nameof#21]
For some reasons, I want to get splitted into the array path instead of string. I mean `'state.field'` becomes  `['state', 'field']`. 
Is it possible to do? Something like `nameof.split<IState>(o => o.state.field);`

**Comment by tankbob** [comment]
Can you not just do nameoffull ().split (".")

**Comment by syxov** [comment]
I can, but I don't want to write it each time

**Comment by dsherret** [comment]
Although I agree that `.split(".")` could just be done, this might as well be implemented.

Signatures:

```
function split<T>(func: (obj: T) => any): string[];
function split(obj: any): string[];
```

**Comment by dsherret** [comment]
This is now implemented in 4.2.0 thanks to @cecilyth! DefinitelyTyped has been updated so the types package is up to date as well and now includes js docs (also thanks to Cecily).
